ccsm-cds-tools/ccsm-cds-with-tests
# CCSMCDS-284: Add data mapping for "Insufficient" histology result

### Summary

Any biopsy with a result SNOMED value of 112631006 should be updated to display as "Insufficient/Unsatisfactory". For now, this probably means adding "Insufficient" as a data element within our CQL, but not actually doing anything with it in terms of the recommendation logic. This is a high priority fix, according to Carolann.

Related JIRA task: https://jira.mitre.org/browse/CCSMCDS-284

### Method

+ Added SNOMED code (112631006)
+ Added handling of the code in cytology
+ Added handling of the code in biopsy

Generated ELM.

### Issue

In the context of `Insufficient` is something handled out of `UNK`, I ranked it the same as `1`. Let me know if this doesn't look right.

### Testing method

1. Add a test case under `Test/cases`. with a YAML below:

---
```
name: I.1 Management of Histologic HSIL, not Further Specified or Qualified 

externalData:
- resources

data:
- 
  resourceType: Patient
  name: Joanne Smith
  gender: female
  birthDate: 1991-01-01
  extension:
  -
    url: http://hl7.org/fhir/us/core/StructureDefinition/us-core-birthsex
    valueCode: F
- 
  resourceType: DiagnosticReport
  code: LOINC#65753-6 Cervix Pathology biopsy report
  status: final
  conclusionCode:
  - SNOMEDCT#112631006 Specimen unsatisfactory for diagnosis (finding)
  effectiveDateTime: 2021-05-01

results:

```

2. Run `test-sql`
3. Since the new code won't make any recommendation, the results should be null and pass as below:

```
CQLT Config: test/Test/cqlt.yaml
I.1 Management of Histologic HSIL, not Further Specified or Qualified: No results specified.
Ignoring potential external data file: /Users/jaehoonlee/Documents/GitHub/ccsm-cds-with-tests/test/Test/cases/resources.yml


  ManageRareAbnormality_v1.1.0
    ✔ I.1 Management of Histologic HSIL, not Further Specified or Qualified (79ms)


  1 passing (196ms)
```
